qutip/solver/heom/bofin_solvers.py

```diff
@@ -753,11 +774,10 @@ def _grad_prev_fermionic(self, he_n, k):
         ]
 
         n_excite = sum(he_fermionic_n)
-        sign1 = (-1) ** (n_excite + 1)
-
```

Comment:
Instead of removing this line and hiding the mathematics, I suggest doing:

`sign1 = (-1) ** (n_excite + 1 - self.odd_partiy)`

and

`sign2 = (-1)**(n_excite_before_m + self.odd_party)`.

This makes the mathematics clearer (I think). It relies on Python numerically treating `False` and `True` as `0` and `1`, which is perhaps worth a comment which I'll suggest further up.

Reply:
You are right I think this indeed clearer